assets/shared/components/modal/index.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * WordPress dependencies
+ */
+import { Icon, close as closeIcon } from '@wordpress/icons';
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Modal component.
+ *
+ * @param {Object}   props
+ * @param {Function} props.onClose  Callback to run when trying to close the modal.
+ * @param {string}   props.title    The title for the modal. Empty by default.
+ * @param {Object}   props.children The content of the modal.
+ * @return {JSX.Element} The modal component or null (if not open).
```

Comment:
I think we don't need the `return` here. At some point, we had decided to not add `@return` to the React components. But if think it's especially important here, feel free to keep it. 😉

Reply:
Removed in here: 1d4574af6